draft-ietf-mls-protocol.md

```diff
@@ -2572,12 +2587,16 @@ message at the same time, by taking the following steps:
   corresponds to the order of PreSharedKey proposals in the `proposals` vector.
   Otherwise, set `psk_secret` to a zero-length octet string.
 
-* Construct an MLSPlaintext object containing the Commit object. Sign the MLSPlaintext
-  using the current epoch's GroupContext as context. Use the signature, the
-  `commit_secret` and the `psk_secret` to advance the key schedule and compute
-  the `confirmation_tag` value in the MLSPlaintext.
+* Use the `init_secret` from the previous epoch, the `commit_secret` and the
+  `psk_secret` as defined in the previous steps, and the new GroupContext to
+  compute the new `joiner_secret`, `welcome_secret`, `epoch_secret`, and
+  derived secrets for the new epoch.
 
-* Update the tree in the provisional state by applying the direct path
+* Construct an MLSPlaintext object containing the Commit object. Sign the
+  MLSPlaintext using the old GroupContext as context. Use the
+  `confirmation_key` for the new epoch to compute the `confirmation_tag` value,
+  and the `membership_key` for the old epoch to compute the `membership_tag`
+  value in the MLSPlaintext.
```

Comment:
I'm not entirely sure why it takes the `confirmation_key` from the new epoch, but the `membership_key` from the old epoch, but that's what the current mlspp code does.

Reply:
Oh, actually, I do know, I think.  The `confirmation_tag` allows the other group members to check that they derived the same tree/secrets as the sender.  The `membership_tag` allows other group members to ensure that it was sent by a group member before processing the message.  I think this is mentioned in other parts of the document, but I can reiterate here if necessary.